### PR TITLE
SNOW-1168205 iceberg table compliance testing

### PR DESCRIPTION
**Pull request by sfc-gh-mhofman**
### Description
Introduction of ICEBERG table CRUD/Structure type/bindings testing

### Checklist
- [x] Code compiles correctly
- [x] Code is formatted according to [Coding Conventions](../blob/master/CodingConventions.md)
- [x] Created tests which fail without the change (if possible)
- [x] All tests passing (`dotnet test`)
- [x] Extended the README / documentation, if necessary
- [x] Provide JIRA issue id (if possible) or GitHub issue id in PR name

### CODE REVIEW

File: Snowflake.Data.Tests/IcebergTests/TestIcebergTable.cs [hunk 1]
```diff
@@ -0,0 +1,562 @@
...
+                CreateIcebergTable(conn);
+                SetResultFormat(conn);
+                InsertManyRowsWithNulls(conn, 20_000, I32, I64, Dec, Dbl, Flt, Txt, B1, B0, _dt, _tm, _ntz, _ltz, _bi);
+
+                // Act
+                var reader = conn.ExecuteReader($"select {SqlColumnsSimpleTypes} from {TableNameIceberg} order by nu1");
+                
+                // Assert
+                var resultSetColumns = SqlCreateIcebergTableColumns.Split('\n');
+                var expected = new Object[] {I32, I64, Dec, Dbl, Flt, Txt, B1, B0, _dt, _tm, _ntz, _ltz, _bi};
```
Comment (sfc-gh-knozderko) [inline Snowflake.Data.Tests/IcebergTests/TestIcebergTable.cs:325]: why Object uppercase?

File: Snowflake.Data.Tests/Snowflake.Data.Tests.csproj [hunk 2]
```diff
@@ -9,7 +9,7 @@
     <Product>Snowflake Connector for .NET</Product>
     <Copyright>Copyright (c) 2012-2023 Snowflake Computing Inc. All rights reserved.</Copyright>
     <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
-    <LangVersion>7.3</LangVersion>
+    <LangVersion>9</LangVersion>
```
Comment (sfc-gh-knozderko) [inline Snowflake.Data.Tests/Snowflake.Data.Tests.csproj:12]: what was it needed to?

**Review by sfc-gh-knozderko — CHANGES_REQUESTED** [review]

File: Snowflake.Data.Tests/IcebergTests/TestIcebergTable.cs [hunk 3]
```diff
@@ -0,0 +1,562 @@
...
+                var index = -1;
+                while (reader.Read())
+                {
+                    ++index;
+                    expected[0] = index+1;
+                    AssertRowValuesEqual(reader, resultSetColumns, NullifyWithinRow(expected, index));
+                }
+            }
+            
+            object[] NullifyWithinRow(object[] expected, int index)
```
[the remark this answers is not in the document]
Comment (sfc-gh-mhofman) [reply Snowflake.Data.Tests/IcebergTests/TestIcebergTable.cs:335]: Fixed parameter name and moved to util class

File: Snowflake.Data.Tests/IcebergTests/TestIcebergTable.cs [hunk 4]
```diff
@@ -0,0 +1,562 @@
...
+                
+                // Assert
+                var reader = conn.ExecuteReader($"select {SqlColumnsSimpleTypes} from {TableNameIceberg}");
+                while (reader.Read())
+                {
+                    AssertRowValuesEqual(reader, SqlCreateIcebergTableColumns.Split('\n'), I32, I64, Dec, Dbl, Flt, Txt, B1, B0, _dt, _tm, _ntz, _ltz, _bi);
+                }
+            }
+
+            string ByteArrayToString(byte[] ba)
```
[the remark this answers is not in the document]
Comment (sfc-gh-mhofman) [reply Snowflake.Data.Tests/IcebergTests/TestIcebergTable.cs:103]: done

**Review by sfc-gh-knozderko — APPROVED** [review]

**Pull request merged**